src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -160,7 +164,9 @@ private void setFilingApiData(
                             String.format("Empty submission returned when generating filing for %s", overseasEntityId)
                     ));
 
-    if (submissionDto.isForUpdate()) {
+    if (submissionDto.isRemoveEntity()) {
+      filing.setKind(FILING_KIND_OVERSEAS_ENTITY_REMOVE);
```

Comment:
We'd probably have to do similar things to an update in this block (like in lines 170 - 173).

Reply:
Added update submission data

Reply:
Thanks. See comment below regarding the 'no change' flag though.